geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/IngestCommand.scala

```diff
@@ -25,14 +28,18 @@ class IngestCommand(parent: JCommander) extends Command(parent) with Logging {
 
   override def execute(): Unit = {
     val fmt = Option(params.format).getOrElse(getFileExtension(params.files(0)))
-    fmt match {
-      case CSV | TSV => new DelimitedIngest(params).run()
-      case SHP       =>
-        val ds = new DataStoreHelper(params).getOrCreateDs
-        GeneralShapefileIngest.shpToDataStore(params.files(0), ds, params.featureName)
-      case _         =>
-        logger.error("Error: File format not supported for file " + params.files(0) + ". Supported formats" +
-          "are csv,tsv,shp")
+    if (fmt == SHP) {
+      val ds = new DataStoreHelper(params).getOrCreateDs()
+      GeneralShapefileIngest.shpToDataStore(params.files(0), ds, params.featureName)
+    } else {
+      val config = Configurator.getConfig(params.config)
+      val sft = Speculator.getSft(params.spec, params.featureName, params.config)
+
+      if (config != null && sft != null) {
+        new DelimitedIngest(params).run()
```

Comment:
should probably pass in the config and sft, or do this check in the delimited ingest class. otherwise the code has to stay in sync between here and there.